File: core-java-modules/core-java-lang-math-3/src/test/java/com/baeldung/magicsquare/ValidationUnitTest.java
package com.baeldung.magicsquare;

import org.junit.jupiter.api.Test;

public class ValidationUnitTest {
    @Test
    public void testValidate3x3() {
        MagicSquare magicSquare = new MagicSquare(3);
        magicSquare.setCell(0, 0, 8);
        magicSquare.setCell(1, 0, 1);
        magicSquare.setCell(2, 0, 6);
        magicSquare.setCell(0, 1, 3);
        magicSquare.setCell(1, 1, 5);
        magicSquare.setCell(2, 1, 7);
        magicSquare.setCell(0, 2, 4);
        magicSquare.setCell(1, 2, 9);
        magicSquare.setCell(2, 2, 2);

        magicSquare.validate();
    }
}

File: core-kotlin/src/test/kotlin/com/baeldung/annotations/ValidationUnitTest.kt
package com.baeldung.annotations

import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Test


class ValidationUnitTest {

    @Test
    fun whenAmountIsOneAndNameIsAlice_thenTrue() {
        assertTrue(Validator().isValid(Item(1f, "Alice")))
    }

    @Test
    fun whenAmountIsOneAndNameIsBob_thenTrue() {
        assertTrue(Validator().isValid(Item(1f, "Bob")))
    }


    @Test
    fun whenAmountIsMinusOneAndNameIsAlice_thenFalse() {
        assertFalse(Validator().isValid(Item(-1f, "Alice")))
    }

    @Test
    fun whenAmountIsMinusOneAndNameIsBob_thenFalse() {
        assertFalse(Validator().isValid(Item(-1f, "Bob")))
    }

    @Test
    fun whenAmountIsOneAndNameIsTom_thenFalse() {
        assertFalse(Validator().isValid(Item(1f, "Tom")))
    }

    @Test
    fun whenAmountIsMinusOneAndNameIsTom_thenFalse() {
        assertFalse(Validator().isValid(Item(-1f, "Tom")))
    }


}